source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/TableCellRenderer.kt

```diff
@@ -0,0 +1,107 @@
+            orientation = LinearLayout.VERTICAL
+            tag = TagContent(cell)
+            layoutParams = TableRow.LayoutParams(0, TableLayout.LayoutParams.MATCH_PARENT)
+        }
+
+        cellLayout.updateLayoutParams<ViewGroup.MarginLayoutParams> {
+            val isFirst = renderArgs.colIndex == 0
+            val isLast = renderArgs.colIndex == renderArgs.table.GetColumns().size - 1
+
+            if (renderArgs.table.GetShowGridLines()) {
+                // For grid lines, draw rectangle
+                cellLayout.background = GradientDrawable().apply {
+                    shape = GradientDrawable.RECTANGLE
+                    setColor(Color.TRANSPARENT)
+
+                    // border color from config
+                    val borderColor = Color.parseColor(hostConfig.GetBorderColor(renderArgs.table.GetGridStyle()))
+                    setStroke(Util.dpToPixels(context, 1f), borderColor)
+                }
+            } else if (cell.GetBleed()) {
+                // To bleed, negate margins of first/last cells
+                val padding = Util.dpToPixels(context, hostConfig.GetSpacing().paddingSpacing.toFloat())
+                if (isFirst) marginStart = -padding
+                if (isLast) marginEnd = -padding
+            } else {
+                // Otherwise, add spacing between cells
+                val spacing = Util.dpToPixels(context, hostConfig.GetTable().cellSpacing.toFloat())
+                if (!isFirst) marginStart = spacing / 2
+                if (!isLast) marginEnd = spacing / 2
+            }
+        }
+
+        if (col.GetWidth() != null) {
+            // Weighted columns can stretch/shrink
+            cellLayout.updateLayoutParams<TableRow.LayoutParams> { width = 0; weight = col.GetWidth()!!.toFloat() }
+            renderArgs.tableLayout.setColumnShrinkable(renderArgs.colIndex, true)
+            renderArgs.tableLayout.setColumnStretchable(renderArgs.colIndex, true)
+        } else if (col.GetPixelWidth() != null) {
+            // Pixel columns are fixed width
+            cellLayout.updateLayoutParams<TableRow.LayoutParams> { width = Util.dpToPixels(context, col.GetPixelWidth()!!.toFloat()) }
+            renderArgs.tableLayout.setColumnShrinkable(renderArgs.colIndex, false)
+            renderArgs.tableLayout.setColumnStretchable(renderArgs.colIndex, false)
+        }
+
+        setVisibility(cell.GetIsVisible(), cellLayout)
+        setMinHeight(cell.GetMinHeight(), cellLayout, context)
+        applyRtl(cell.GetRtl(), cellLayout)
+
+        val computedStyle = ContainerRenderer.getLocalContainerStyle(cell, renderArgs.containerStyle)
+        ContainerRenderer.applyPadding(computedStyle, renderArgs.containerStyle, cellLayout, hostConfig, renderArgs.table.GetShowGridLines())
+        ContainerRenderer.applyContainerStyle(computedStyle, cellLayout, hostConfig)
+        ContainerRenderer.applyVerticalContentAlignment(cellLayout, cell.GetVerticalContentAlignment()
```

Comment:
a comment would be nice here explaining the precedence between the verticalcontentalignment assigned through elvis operators